### DISCUSSION TITLE
Possible to access desktop version from browser?

### DISCUSSION THREAD

**Discussion by LorenAmelang** [TriliumNext/Trilium#2443] [Q&A]
This may be a silly idea, but I wanted to see what my iPad might show if I set up a server. So I sent Safari to the IP of my running desktop install, and the port the web clipper showed it was connected to. And I get the Trilium icon! Plus some "Unauthorized" errors:

![Trilium Desktop](https://user-images.githubusercontent.com/1297090/146136016-ef249769-fce1-4376-8547-f05abb081871.png)

Same result from a desktop browser on the machine running Trilium. Obviously they are connecting to something...  Might there be some way to "authorize" this connection?

**Comment by zadam** [comment]
Hi, you can disable this protection in `config.ini` file:

```
# set to true to allow using Trilium without authentication (makes sense for server build only, desktop build doesn't need password)
noAuthentication=true
```

Smoke test works, but there might be some weird things since this was never tested. Also desktop builds don't contain mobile interface so only full desktop one is available.

**Comment by LorenAmelang** [comment]
My config.ini didn't have that line, I added it. With Trilium closed. Ran it again, same errors in browser. The "port 8080" mentioned in the config refuses a connection...  

First try put this in the log:

15:26:39.998 JS Error: Uncaught error: Message: Uncaught TypeError: Cannot read properties of undefined (reading 'protectedSessionTimeout'), URL: http://10.1.1.10:37840/app-dist/desktop.js, Line: 2, Column: 70270, Error object: {}, Stack: TypeError: Cannot read properties of undefined (reading 'protectedSessionTimeout')

Later tries don't repeat that. 

Maybe I've missed some detail? 

Verbose config and logs in the file here... 
[auth test.txt](https://github.com/zadam/trilium/files/7723607/auth.test.txt)

**Reply by zadam** [reply]
It has to be in "General" section of config.ini, I forgot to add this detail. Actual doc is [here](https://github.com/zadam/trilium/wiki/Server-installation#disable-authentication)

Also, desktop builds don't respect port settings (8080), instead it is fixed to 37840.

**Reply by LorenAmelang** [reply]
That Works! Beautifully! Now I really want to set up a server...  

One little question, that I can't find mentioned anywhere. I noticed the instanceName setting, and it seemed it might be useful when I have access from several sources. 

```
[General]
# Instance name can be used to distinguish between different instances
instanceName=SB
```
I don't see any effect, in the display, the GUI Settings, or the log. What should it do? Where might it be visible? 

Maybe it needs delimiters in the config file? But pathnames don't in the examples I found.

**Reply by zadam** [reply]
The only usage is in scripting which allows creating conditional logic based on the instance name.
